tests/test_tag_item.py

```diff
@@ -0,0 +1,18 @@
+#-*- coding: utf-8 -*-
+import logging
+import sys
+
+import pytest
+import rasterio
+
+logging.basicConfig(stream=sys.stderr, level=logging.DEBUG)
+
+
+def test_get_tag_item():
+    with rasterio.open('tests/data/cogeo.tif') as src:
+        assert src.get_tag_item('INTERLEAVE', 'IMAGE_STRUCTURE') == 'PIXEL'
+        assert src.get_tag_item('IFD_OFFSET', 'TIFF', bidx=1) == '8'
+        assert src.get_tag_item('IFD_OFFSET', 'TIFF', bidx=1, ovr=1) == '1504'
+        assert not src.get_tag_item('IF', 'TIFF', bidx=1)
+        with pytest.raises(Exception):
+            src.get_tag_item('IFD_OFFSET', 'TIFF', ovr=1)
```

Comment:
@sgillies in this commit I added some code to make this method ~= to `ds.tags` method, meaning it still returns tag value but it should also work on the whole dataset (` src.get_tag_item('INTERLEAVE', 'IMAGE_STRUCTURE')`)

Reply:
I would prefer not to have 2 different ways of getting `IMAGE_STRUCTURE:INTERLEAVE`.

Reply:
I understand. 
IMO it made sense that `src.get_tag_item` should be an enhanced version of `src.tags` (as GDALGetMetadataItem is for GDALGetMetadata). 

I could remove it (meaning `src.get_tag_item` will work only on bands object) but I fear it could create some confusion.

Reply:
Okay! I'm convinced we can handle overlapping methods with documentation. I retract my change request :)